src/Storage/Services/ApplicationService.cs

```diff
@@ -38,5 +42,38 @@ public ApplicationService(IApplicationRepository applicationRepository)
 
             return (false, new ServiceError(405, $"DataType {dataType} is not declared in application metadata for app {appId}"));
         }
+
+        /// <summary>
+        /// Wrapper method for getting an application or an error message using the application id.
+        /// </summary>
+        /// <param name="appId">The application id</param>
+        /// <returns></returns>
+        public async Task<(Application Application, ActionResult ErrorMessage)> GetApplicationOrErrorAsync(string appId)
+        {
+            ActionResult errorResult = null;
+            Application appInfo = null;
+
+            try
+            {
+                string org = appId.Split("/")[0];
+
+                appInfo = await _applicationRepository.FindOne(appId, org);
+
+                if (appInfo == null)
+                {
+                    errorResult = new NotFoundObjectResult($"Did not find application with appId={appId}");
```

Comment:
I want us to avoid getting new API related dependencies down into the service (core/domain) layer. Use the ServiceError class and translate that into a response in the controller.